### PR TITLE
Feat: InstantClick & DNS Prefetch

### PR DESCRIPTION
**Pull request by frank-xjh**
首先，十分感谢你花时间来给 OI Wiki 开一个 Pull Request，下面是一些你可能需要知道的信息：

- 请在 commit 的时候写比较有意义的 commit message
- 请给 PR 起比较有意义的标题。
- 请在 PR 之前检查一下您的 PR 是否存在以下常见问题：（确认无问题后请将选项打钩 / 填为 `[x]`）
   * [ ] 您的 MD 代码的书写格式，包括但不限于 **中文与英文之间、中文与阿拉伯数字、中文与 LaTeX 公式之间要有一个半角空格**，特别地，在中文全角符号与英文、阿拉伯数字、LaTeX 公式之间，**不需要**半角空格。（这个可以使用自动化工具辅助，比如 https://github.com/baurine/vscode-pangu)
   * [ ] 对于 LaTeX 公式，请注意常见的问题，**一定要使用** `$\log$`、`$\min$`、`$\max$`、`$\gcd$` 等，而非 `$log$`、`$min$`、`$max$`、`$gcd$`。对于最小公倍数，请使用 `$\operatorname{lcm}$` 而非 `$lcm$`，省略号请使用 `\cdots`，叉乘请使用 `\times`，点乘请使用 `\cdot`。
   * [ ] 所有公式中的希腊字母等特殊符号，请不要使用输入法的插入特殊符号功能，而应该使用对应的 LaTeX 公式符号。如 `ϕ` 应该使用 `$\phi$` 或 `\varphi`。
   * [ ] 行间公式前后各要有一行空行。
   * [ ] 对于目录中存在公式的情况，请使用 HTML 代码而非直接插入公式以避免双倍公式的问题，请参考[避免 ToC 中双倍公式的写法](https://oi-wiki.org/intro/faq/#_13))。

关于文档内容的基本格式可以查阅 [F.A.Q](https://oi-wiki.org/intro/faq/#_4)。

### CODE REVIEW

**Comment by frank-xjh** [comment]
server好在这：https://test.4leaf.top/OI-Wiki/

File: static/footer.html [hunk 1]
```diff
@@ -22,6 +22,45 @@
...
+    jax: ["input/TeX", "output/HTML-CSS"],
+    tex2jax: {
+        inlineMath:  [ ["$", "$"] ],
+        displayMath: [ ["$$","$$"] ],
+        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre','code','a'],
+        ignoreClass:"comment-content"
+    },
+    "HTML-CSS": {
+        availableFonts: ["STIX","TeX"],
+        showMathMenu: false
```
Comment (Ir1d) [inline static/footer.html]: 这个为啥要关掉

File: static/footer.html [hunk 2]
```diff
@@ -22,6 +22,45 @@
...
+<script src="//cdn.bootcss.com/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
+
+<!-- InstantClick -->
+<script type="text/javascript" src="https://cdn.bootcss.com/instantclick/3.0.0/instantclick.js" data-no-instant></script>
+<script data-no-instant>
+InstantClick.on('change', function(isInitialLoad) {
+ if (isInitialLoad === false) {
+  if (typeof MathJax !== 'undefined') // support MathJax
+   MathJax.Hub.Queue(["Typeset",MathJax.Hub]);
+  if (typeof prettyPrint !== 'undefined') // support google code prettify
```
Comment (Ir1d) [inline static/footer.html]: 这个以及百度统计都可以去掉吧？

File: mkdocs.yml [hunk 3]
```diff
@@ -239,7 +239,6 @@ extra:
 extra_javascript:
   - 'https://oi-wiki.org/analytics.js'
   - '_static/js/extra.js?v=2'
```
Comment (Ir1d) [inline mkdocs.yml]: 改成 `v=3` 吧

File: static/footer.html [hunk 4]
```diff
@@ -22,6 +22,45 @@
 
 {% import "partials/language.html" as lang with context %}
 
+<script type="text/x-mathjax-config">
+MathJax.Hub.Config({
+    showProcessingMessages: false,
+    messageStyle: "none",
+    extensions: ["tex2jax.js"],
+    jax: ["input/TeX", "output/HTML-CSS"],
```
Comment (Ir1d) [inline static/footer.html]: 这个换成 CommonHTML 怎么样？我本地试了下可以用（就是好像不支持自定义字体了）

ref：
http://docs.mathjax.org/en/latest/output.html#mathjax-output-components
> Its major advantage is its quality, consistency, and speed as well as support for server-side generation
https://groups.google.com/forum/#!topic/mathjax-users/Mm_pw4_cZgA

Comment (frank-xjh) [reply static/footer.html]: 可以换

**Comment by frank-xjh** [comment]
添加DNS Prefetch

File: build.sh [hunk 5]
```diff
@@ -15,6 +15,7 @@ if [ ! -d "mkdocs-material" ] ; then
 fi
 sed -i "s/name: 'material'/name: null\n  custom_dir: 'mkdocs-material\/material'\n  static_templates:\n    - 404.html/g" mkdocs.yml
 
+sed '9 a\<meta http-equiv="x-dns-prefetch-control" content="on">\n\<link rel="dns-prefetch" href="//fonts.loli.net">\n\<link rel="dns-prefetch" href="//cdn.bootcss.com">\n\<link rel="dns-prefetch" href="//unpkg.com">\n\<link rel="dns-prefetch" href="//npmcdn.com">\n\<link rel="dns-prefetch" href="//oi-wiki.org">\n\<link rel="dns-prefetch" href="//api.github.com">\n\<link rel="dns-prefetch" href="//www.google-analytics.com">' mkdocs-material/material/base.html
```
Comment (Ir1d) [inline build.sh]: `sed -i` ?

Comment (frank-xjh) [reply build.sh]: 再第九行后插入

**Comment by Ir1d** [comment]
@frank-xjh 可能类似这样？我找不到上回测试的了
```
MathJax.Hub.Config({
    showProcessingMessages: false,
    messageStyle: "none",
    extensions: ["tex2jax.js"],
    jax: ["input/TeX", "output/CommonHTML"],
    tex2jax: {
        inlineMath:  [ ["$", "$"] ],
        displayMath: [ ["$$","$$"] ],
        skipTags: ['script', 'noscript', 'style', 'textarea', 'pre','code','a'],
        ignoreClass:"comment-content"
    },
});
MathJax.Hub.Queue(["Typeset",MathJax.Hub]);

```

http://docs.mathjax.org/en/latest/options/output-processors/CommonHTML.html#configure-commonhtml
